src/common/utils/criteria.py

```diff
@@ -0,0 +1,92 @@
+from django.contrib.contenttypes.models import ContentType
+from common.models import NestedCriteriaGroup, NestedCriteria
+
+
+def get_nested_criteria(model, id):
+    """
+    Retrieve nested criteria for a given model and id.
+
+    This function fetches the nested criteria associated with a specific model and id.
+    It first determines the content type of the model, then retrieves the parent group
+    of nested criteria. If a parent group exists, it gathers the nested group data and
+    appends it to the nested_criteria list.
+
+    Args:
+        model (Model): The Django model for which nested criteria are to be retrieved.
+        id (int): The ID of the specific instance of the model.
+
+    Returns:
+        list: A list of nested criteria data.
+    """
+    nested_criteria = []
+
+    # get the content type of the model
+    content_type = ContentType.objects.get_for_model(model)
+
+    # get the parent group
+    parent_group = NestedCriteriaGroup.objects.filter(
+        content_type=content_type, object_id=id, parent_group__isnull=True
+    )
+
+    if parent_group.count() != 0:
+        for group in parent_group:
+            group_data = get_nested_group_data(group)
+            nested_criteria.append(group_data)
+
+    return nested_criteria
+
+
+def get_nested_group_data(group_instance):
+    """
+    Recursively retrieves and constructs a nested dictionary representation of a group instance and its nested groups.
+
+    Args:
+        group_instance (Group): The group instance for which to retrieve nested group data.
+
+    Returns:
+        dict: A dictionary containing the group's id, operator, criteria, and any inner nested groups.
+    """
+    group_data = {
+        "id": group_instance.id,
+        "operator": group_instance.operator,
+        "criteria": get_nested_criteria_for_group(group_instance),
+        "innerGroups": [],
```

Comment:
Any reason we're using `camelCase` naming here?

Reply:
This data directly corresponds to the frontend data, with a similar structure. To avoid redundant mapping on the frontend, I used similar variable names, following the camelCase convention used on the frontend.